website/blog/2021-02-19-tungsten.md

```diff
@@ -0,0 +1,498 @@
+Sublime Text and Eclipse. Metals is developed at the
+[Scala Center](https://scala.epfl.ch/) and [VirtusLab](https://virtuslab.com)
+with the help from [Lunatech](https://lunatech.com) along with contributors from
+the community.
+
+## TL;DR
+
+Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
+give Metals a try!
+
+- Type decorations for definitions.
+- Code action to insert the inferred type annotation.
+- Configuration option for the default Scala version.
+- Code lens to easily discover copy worksheet command.
+- Improved short names when importing.
+- Debug run/test when using sbt via BSP.
+- Support for Scala 3.0.0-RC1
+- [Scala 3] Filtering and sorting for completions.
+- [Scala 3] Basic shorter signature support for hover.
+- [Scala 3] Document outline
+- [Scala 3] Code folding
+- [Scala 3] Formatting `|` in multiline string
+- [Scala 3] Code action to convert to multiline string.
+
+## Type decorations for definitions.
+
+One of the most
+[requested features](https://github.com/scalameta/metals/issues/707) for Metals
+was the ability to show the inferred type for anything that does not have it
+defined explicitly. This includes the type at:
+
+- value and variable definitions
+
+```scala
+val a = 1
+var b = 2
+```
+
+- method definitions
+
+```scala
+def method() = ""
+```
+
+- variables in pattern matches
+
+```scala
+list match {
+  case aa :: tail =>
+}
+```
+
+- lambda parameters
+
+```scala
+list.map(param => param.toInt)
+```
+
+- for comprehensions
+
+```scala
+for(i <- 0 to 10)`
+```
+
+The decoration is based on the information produced during the compilation, so
+anything new will first need to be compiled to have the type shown.
+
+![type-decorations](https://i.imgur.com/NHzB0M6.png)
+
+This features can easily be controlled by the previously available
+`metals.showInferredType` setting.
+
+## Code action to insert inferred type.
+
+Another highly requested feature and closely related to the type decorations is
+the ability to automatically insert type annotation. This is achieved using a
+new `Insert type annotations` and
+`Insert type annotation into pattern definition` code actions. Those actions
+work in the same places as mentioned in the previous point - method, variables,
+lambdas etc. - but uses the presentation compiler instead of indexes. What it
+means is that this new feature will work even in some more problematic cases,
+where the file does not yet compiled or saved and files that type decorations
+are not yet supported like in for example worksheets.
+
+![inferred-type](https://i.imgur.com/ZT36mH7.gif)
+
+There might be still some work to do when it comes to some more complicated
+types, but it should work perfectly well for most daily use cases. Please do
+report any issues you might encounter, it will help us greatly to polish
+printing the types properly.
+
+The feature will currently work for clients that allow virtual text to be
```

Comment:
Is this feature really dependent on the ability to show decorations ? Isn't code actions support from the client enough ?

Reply:
Good catch :O That was supposed to be mentioned for decorations